lib/xenops_server.ml

```diff
@@ -40,6 +40,8 @@ let query _ _ _ = {
   features = [];
   instance_id = instance_id;
 }
+let local_query = query ()()()
+let key_query_of_sender = "query_of_sender"
```

Comment:
I think the name of this cookie is a bit mysterious. Why not something like "vgpu-migration" and thus revealing that this is about features?

Reply:
Why not call it `version` and use `version = 2` for new migrations?

Reply:
Is this a key/value pair? If so, I think we should use the value and not just use the key.

Reply:
We do `List.assoc` on it later, so I guess it is. And using version would allow us to update the protocol meaningfully in the future if needed (say for other vendors VGPU migrations for example)

Reply:
Initially I thought about only sending only a new "allow_vgpu_migration" cookie with an empty value, but then I realised that it would be more useful to send some extra information from the sending host, which is available in the local_query value. I was hoping that in the future this query cookie that comes from the sending host could be used by the receiving host to decide to accept or not the migration using the query's fields such as vendor, version and features. It just happens that at this point in which this cookie is introduced, it's being used by the receiver as a way to prevent the vgpu-migration, but in the future it could be used as well to prevent any other actions during migration. Maybe we should just simplify this idea and send instead the "allow_vgpu_migration" cookie?